app/cli/main.go

```diff
@@ -16,25 +16,82 @@
 package main
 
 import (
+	"crypto/sha256"
+	"fmt"
 	"os"
+	"strings"
 
 	"github.com/chainloop-dev/chainloop/app/cli/cmd"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry/posthog"
 	v1 "github.com/chainloop-dev/chainloop/app/controlplane/api/controlplane/v1"
 	"github.com/go-kratos/kratos/v2/errors"
 	jwtMiddleware "github.com/go-kratos/kratos/v2/middleware/auth/jwt"
 	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	"google.golang.org/grpc/status"
 )
 
+// Follow the convention stated on https://consoledonottrack.com/
+const doNotTrackEnv = "DO_NOT_TRACK"
+
+var (
+	// Posthog API key and endpoint to be injected in the build process
+	posthogAPIKey   = ""
+	posthogEndpoint = ""
+)
+
 func main() {
 	// Couldn't find an easier way to disable the timestamp
 	logger := zerolog.New(zerolog.ConsoleWriter{Out: os.Stderr, FormatTimestamp: func(interface{}) string { return "" }})
 	rootCmd := cmd.NewRootCmd(logger)
 
-	if err := rootCmd.Execute(); err != nil {
-		msg, exitCode := errorInfo(err, logger)
+	// Set the default exit code to 0, in case of an error it will be updated
+	exitCode := 0
+
+	// Run the command and get the executed command
+	executedCmd, err := rootCmd.ExecuteC()
+	if err != nil {
+		var msg string
+		msg, exitCode = errorInfo(err, logger)
 		logger.Error().Msg(msg)
-		os.Exit(exitCode)
+	}
+
+	runTelemetry(logger, executedCmd, exitCode)
+
+	os.Exit(exitCode)
+}
+
+// runTelemetry sends the telemetry data to the configured endpoint
+func runTelemetry(logger zerolog.Logger, executedCmd *cobra.Command, exitCode int) {
+	var telemetryClient telemetry.Client
+
+	if !isTelemetryDisabled() {
```

Comment:
tbh, at this point is probably better that we wrap this `runTelemetry` method within `!isTelemetryDisabled`, that way we do not run any useless compute like generating all those tags and so on if we know it's not going to happen.

In other words, I'd do something like

if !isTelemetryDisabled() { runTelemetry(...) }